### PR TITLE
[KOGITO-7922] - Update Kogito Operator fo Go 1.17(#1280)

### PR DESCRIPTION
**Pull request by spolti**
Cherry-picks https://github.com/kiegroup/kogito-operator/pull/1280

Please make sure your PR meets the following requirements:

- [ ] You have read the [contributors' guide](https://github.com/kiegroup/kogito-operator/blob/main/README.md#contributing-to-the-kogito-operator)
- [ ] Pull Request title is properly formatted: `[KOGITO-XYZ] Subject`
- [ ] Pull Request contains a link to the JIRA issue
- [ ] Pull Request contains a description of the issue
- [ ] Pull Request does not include fixes for issues other than the main ticket
- [ ] Your feature/bug fix has a unit test that verifies it
- [ ] You've ran `make before-pr` and everything is working accordingly
- [ ] You've tested the new feature/bug fix in an actual OpenShift cluster
- [ ] You've added a [RELEASE_NOTES.md](https://github.com/kiegroup/kogito-operator/blob/main/RELEASE_NOTES.md) entry regarding this change

<details>
<summary>
How to retest this PR or trigger a specific build:
</summary>

* <b>Run operator BDD testing</b>
  Please add comment: <b>/jenkins test</b>

* <b>Run RHPAM operator BDD testing</b>
  Please add comment: <b>/jenkins RHPAM test</b>
</details>

### CODE REVIEW

File: .ci/jenkins/Jenkinsfile.deploy [hunk 1]
```diff
@@ -10,9 +10,9 @@ pipeline {
 
     // Needed for local build
     tools {
-        jdk 'kie-jdk11'
-        maven 'kie-maven-3.8.1'
-        go 'golang-1.16'
+        maven env.BUILD_MAVEN_TOOL
+        jdk env.BUILD_JDK_TOOL
```
Comment (radtriste) [inline .ci/jenkins/Jenkinsfile.deploy:14]:
```suggestion
        jdk 'kie-jdk11'
        maven 'kie-maven-3.8.1'
```

File: .ci/jenkins/Jenkinsfile.profiling [hunk 2]
```diff
@@ -9,9 +9,9 @@ pipeline {
 
     // Needed for local build
     tools {
-        jdk 'kie-jdk11'
-        maven 'kie-maven-3.8.1'
-        go 'golang-1.16'
+        maven env.BUILD_MAVEN_TOOL
+        jdk env.BUILD_JDK_TOOL
```
Comment (radtriste) [inline .ci/jenkins/Jenkinsfile.profiling:13]:
```suggestion
        jdk 'kie-jdk11'
        maven 'kie-maven-3.8.1'
```

File: README.md [hunk 3]
```diff
@@ -137,12 +137,12 @@ For code contributions, review the following prerequisites:
...
-- [Operator SDK](https://github.com/operator-framework/operator-sdk) v1.11.0
-- [Go](https://golang.org/) v1.16 is installed.
-- [Golint dependency](https://pkg.go.dev/golang.org/x/lint/golint): go get -u golang.org/x/lint/golint
+- [Operator SDK](https://github.com/operator-framework/operator-sdk) v1.21.0
+- [Go](https://golang.org/) v1.17 is installed.
+- [Golint dependency](https://pkg.go.dev/golang.org/x/lint/golint): go install golang.org/x/lint/golint@latest
 - [Golangci-lint](https://golangci-lint.run/usage/install/)
 - [Python 3.x](https://www.python.org/downloads/) v3.x is installed
-- [Cekit](https://cekit.io/) 4.1.1+ is installed
+- [Cekit](https://cekit.io/) v4.0.0+ is installed
```
Comment (radtriste) [inline README.md:145]:
```suggestion
- [Cekit](https://cekit.io/) v4.1.1+ is installed
```

Comment (spolti) [reply README.md:145]: I kept what was on main branch.
Better to update it later, no?

File: hack/addheaders.sh [hunk 4]
```diff
@@ -13,6 +13,6 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-command -v  addlicense > /dev/null || go get -modfile=go.tools.mod -u github.com/google/addlicense@v1.0.0
+command -v  addlicense > /dev/null || go install -modfile=go.tools.mod github.com/google/addlicense@v1.0.0
```
Comment (radtriste) [inline hack/addheaders.sh:16]:
```suggestion
command -v  addlicense > /dev/null || go install github.com/google/addlicense@v1.0.0
```

File: hack/go-lint.sh [hunk 5]
```diff
@@ -13,7 +13,7 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-command -v golint > /dev/null || go get -u -modfile=go.tools.mod golang.org/x/lint/golint
+command -v golint > /dev/null || go install -modfile=go.tools.mod golang.org/x/lint/golint@latest
```
Comment (radtriste) [inline hack/go-lint.sh:16]:
```suggestion
command -v golint > /dev/null || go install golang.org/x/lint/golint@latest
```

Comment (spolti) [reply hack/go-lint.sh:16]: we can probably remove the go.tools.mod and sum as well

File: README.md (hunk 3 above)
Comment (radtriste) [reply README.md:145]: oki

**Review by sutaakar — APPROVED** [review]

**Review by radtriste — APPROVED** [review]
Can be merged once product check is good

**Pull request merged**